tests/test_BioLogic.py

```diff
@@ -358,3 +377,25 @@ def test_MPR_matches_MPT_v1150(testdata_dir, basename_v1150):
     mpr = MPRfile(binpath)
     mpt, comments = MPTfile(txtpath, encoding="latin1")
     assert_MPR_matches_MPT_v2(mpr, mpt, comments)
+
+
+def test_loop_from_file(testdata_dir):
+    """Check if the loop_index is correctly extracted from the _LOOP.txt file
+    """
+    mpr = MPRfile(os.path.join(testdata_dir, "running", "running_OCV.mpr"))
+    if mpr.loop_index is None:
+        raise AssertionError("No loop_index found")
+    elif not len(mpr.loop_index) == 4:
+        raise AssertionError("loop_index is not the right size")
+    elif not (mpr.loop_index == [0, 4, 8, 11]).all():
+        raise AssertionError("loop_index values are wrong")
+
+
+def test_timestamp_from_file(testdata_dir):
+    """Check if the loop_index is correctly extracted from the _LOOP.txt file
+    """
+    mpr = MPRfile(os.path.join(testdata_dir, "running", "running_OCV.mpr"))
+    if not hasattr(mpr, "timestamp"):
+        raise AssertionError("No timestamp found")
+    elif not mpr.timestamp.timestamp() == 1707299985.908:
```

Comment:
With floating point numbers you need to use `assert_close()` or similar, because they will not always be exactly equal.